### PR TITLE
MM-46864 : Migrate "actions/team_actions.js" and tests to typescript

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Migrated "action/team_actions.js" to typescript. 

🔗 This is part of effort for adding graphQL to channels

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46864

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/11062 has to be merged first.

#### Screenshots
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]
LGTM, thanks @M-ZubairAhmed !

File: actions/team_actions.test.ts [hunk 1]
```diff
@@ -74,27 +74,28 @@ jest.mock('utils/browser_history', () => ({
...
+                manuallyUnread: {},
             },
-        });
-    });
+        },
+    };
 
     describe('switchTeam', () => {
         test('should switch to a team by its URL if no team is provided', () => {
+            const testStore = configureStore(initialState);
```
Comment (m1lt0n) [inline actions/team_actions.test.ts:88]: (question) out of curiosity, why the change from `beforeEach` to configuring the store in each test? I see all tests configure the store similarly (ie no test needs a different initial state eg)?

**Review by m1lt0n — APPROVED** [review]
Looks good Zubair! Left one non-blocking comment/question

File: actions/team_actions.test.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply actions/team_actions.test.ts:88]: The before would have worked as expected but i just have to write it like
```
    let testStore: ReturnType<typeof configureStore>;
    const initialState = {
        entities: {
            channels: {
                currentChannelId,
                manuallyUnread: {},
            },
        },
    };
    beforeEach(async () => {
        testStore = configureStore({
            entities: {
                channels: {
                    currentChannelId,
                    manuallyUnread: {},
                },
            },
        });
    });
 ```
 
 But i think its better to have separate store in each test case
